lib/rules/no-java-constructor.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * @fileoverview Rule to warn about declaring a method with the same name of the class. The
+ * developer was likely writing Java recently and meant to write `constructor`.
+ * @author Michael Bolin
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "disallow method names that match the class name",
+            category: "Possible Errors",
+            recommended: true
```

Comment:
We can't enable rules in `eslint:recommended` until the next major release, so please set this to `false` for now.